security/keycloak-oidc-client-extended/src/test/java/io/quarkus/ts/security/keycloak/oidcclient/extended/restclient/DpopIT.java

```diff
@@ -0,0 +1,198 @@
+package io.quarkus.ts.security.keycloak.oidcclient.extended.restclient;
+
+import static io.quarkus.test.bootstrap.KeycloakService.DEFAULT_REALM;
+import static io.quarkus.test.bootstrap.KeycloakService.DEFAULT_REALM_BASE_PATH;
+import static io.quarkus.test.bootstrap.KeycloakService.DEFAULT_REALM_FILE;
+import static io.restassured.RestAssured.given;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.oidc.common.runtime.OidcCommonUtils;
+import io.quarkus.oidc.runtime.OidcUtils;
+import io.quarkus.test.bootstrap.KeycloakService;
+import io.quarkus.test.bootstrap.Protocol;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.KeycloakContainer;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.Response;
+import io.smallrye.jwt.build.Jwt;
+import io.smallrye.jwt.build.JwtClaimsBuilder;
+import io.smallrye.jwt.build.JwtSignatureBuilder;
+import io.smallrye.jwt.util.KeyUtils;
+
+@QuarkusScenario
+@Tag("https://issues.redhat.com/browse/QUARKUS-5858")
+public class DpopIT {
+    private static final String USERNAME = "test-user";
+    private static final String PASSWORD = "test-user";
+
+    private static final String CLIENT_ID = "test-application-client";
+    private static final String CLIENT_SECRET = "test-application-client-secret";
+
+    @KeycloakContainer(command = { "start-dev", "--import-realm",
```

Comment:
In OpenShift, we need to use `image = "${rhbk.image}"` because IBM Z/PE fail with upstream image (architecture not supported). We could also disable it for them, but seems unnecesary.

Reply:
I've added this for OCP tests